### PR TITLE
Added tile function to ValueArray and Matrix classes.

### PR DESCRIPTION
**Pull request by loudinb**
Tile function was added to ValueArray and Matrix classes.  The tile
function generate a Matrix by tiling the source (ValueArray or Matrix)
MxN times.

### CODE REVIEW

File: Source/1D/ValueArray.swift [hunk 1]
```diff
@@ -202,6 +202,25 @@ open class ValueArray<Element: Value>: MutableLinearType, ExpressibleByArrayLite
     open func toColumnMatrix() -> Matrix<Element> {
         return Matrix(rows: count, columns: 1, elements: self)
     }
+    
+    open func toMatrix(rows: Int, columns:Int) -> Matrix<Element> {
+        precondition(rows*columns == count, "Element count must equal rows*columns")
+        return Matrix(rows: rows, columns: columns, elements: self)
+    }
+    
+    open func tile(m: Int, n: Int) -> Matrix<Element> {
```
Comment (alejandro-isaza) [inline Source/1D/ValueArray.swift]: I would remove the argument names and have instead `tile(_ m: Int, _ n Int)`

File: Source/2D/Matrix.swift [hunk 2]
```diff
@@ -206,6 +206,26 @@ open class Matrix<Element: Value>: MutableQuadraticType, Equatable, CustomString
 
         return description
     }
+    
+    open func tile(m: Int, n: Int) -> Matrix {
```
Comment (alejandro-isaza) [inline Source/2D/Matrix.swift]: Same here

File: README.md [hunk 3]
```diff
@@ -84,6 +84,25 @@ let B = inv(A) * C // [2.0, 1.0]′
...
+### Tiling
+A block  `Matrix`  can be formed by repeating a 1-D `ValueArray` or 2-D `Matrix`  **mxn** times.
+
+```swift
+import Upsurge
+
+let a = ValueArray = [1.0, 2.0]
+// Tile source array 2 times in each directon,
+// returning a 2X4 block matrix
+let A = a.tile(m: 2, n:2)
```
Comment (alejandro-isaza) [inline README.md]: I think `let A = a.tile(2, 2)` is a bit more clear.

**Review by alejandro-isaza — CHANGES_REQUESTED** [review]

**Comment by loudinb** [comment]
Thank you @aleph7 .  I have modified and committed the changes.

**Review by alejandro-isaza — APPROVED** [review]

**Comment by alejandro-isaza** [comment]
Thanks!

**Pull request merged**